tests/functional/init/test_init.py

```diff
@@ -0,0 +1,688 @@
+      port: [port]
+      user: [prod_username]
+      pass: [prod_password]
+      dbname: [dbname]
+      schema: [prod_schema]
+
+  target: dev
+"""
+            )
+
+
+class TestInitProjectWithProfileTemplateWithoutExistingProfilesYml:
+    @mock.patch("dbt.task.init._get_adapter_plugin_names")
+    @mock.patch("click.confirm")
+    @mock.patch("click.prompt")
+    @mock.patch.object(Path, "exists", autospec=True)
+    def test_init_task_in_project_with_profile_template_without_existing_profiles_yml(
+        self, exists, mock_prompt, mock_confirm, mock_get_adapter, project
+    ):
+        def exists_side_effect(path):
+            # Override responses on specific files, default to 'real world' if not overriden
+            return {
+                "profiles.yml": False,
+            }.get(path.name, os.path.exists(path))
+
+        exists.side_effect = exists_side_effect
+
+        with open("profile_template.yml", "w") as f:
+            f.write(
+                """fixed:
+  type: postgres
+  threads: 4
+  host: localhost
+  dbname: my_db
+  schema: my_schema
+  target: my_target
+prompts:
+  target:
+    hint: 'The target name'
+    type: string
+  port:
+    hint: 'The port (for integer test purposes)'
+    type: int
+    default: 5432
+  user:
+    hint: 'Your username'
+  pass:
+    hint: 'Your password'
+    hide_input: true"""
+            )
+
+        manager = Mock()
+        manager.attach_mock(mock_prompt, "prompt")
+        manager.attach_mock(mock_confirm, "confirm")
+        manager.prompt.side_effect = ["my_target", 5432, "test_username", "test_password"]
+        mock_get_adapter.return_value = [project.adapter.type()]
+        run_dbt(["init"])
+        manager.assert_has_calls(
+            [
+                call.prompt(
+                    "target (The target name)", default=None, hide_input=False, type=click.STRING
+                ),
+                call.prompt(
+                    "port (The port (for integer test purposes))",
+                    default=5432,
+                    hide_input=False,
+                    type=click.INT,
+                ),
+                call.prompt("user (Your username)", default=None, hide_input=False, type=None),
+                call.prompt("pass (Your password)", default=None, hide_input=True, type=None),
+            ]
+        )
+
+        with open(os.path.join(project.profiles_dir, "profiles.yml"), "r") as f:
+            assert (
+                f.read()
+                == """test:
+  outputs:
+    my_target:
+      dbname: my_db
+      host: localhost
+      pass: test_password
+      port: 5432
+      schema: my_schema
+      threads: 4
+      type: postgres
+      user: test_username
+  target: my_target
+"""
+            )
+
+
+class TestInitInvalidProfileTemplate:
+    @mock.patch("dbt.task.init._get_adapter_plugin_names")
+    @mock.patch("click.confirm")
+    @mock.patch("click.prompt")
+    def test_init_task_in_project_with_invalid_profile_template(
```

Comment:
So this test would get the `profiles.yml` that exisits in `tests/functional/init`?

Reply:
This test would have a profiles.yml created for the `project` fixture in `{project.project_root}/{project.profiles_dir}/profiles.yml` to start, which gets overwritten by invoking `dbt init` during the test.